spark_connect_labextension/sparkconnectserver/cluster.py

```diff
@@ -0,0 +1,227 @@
+"""
+Spark Connect cluster connection
+
+This file contains a singleton object `cluster` to start, stop, and get the status of the Spark Connect server process.
+"""
+
+import os
+import subprocess
+import tempfile
+import glob
+import socket
+from string import Formatter
+from enum import Enum
+from spark_connect_labextension.config import SPARK_HOME, SPARK_CONNECT_PORT, SPARK_CONNECT_PACKAGE
+
+
+class ClusterStatus(Enum):
+    """
+    Enum for Spark cluster connection status
+    """
+    STOPPED = "STOPPED"
+    PROVISIONING = "PROVISIONING"
+    READY = "READY"
+
+
+class _SparkConnectCluster:
+    """
+    Spark Connect server model object.
+    This class contains methods to start, stop, and get the state of the Spark Connect server.
+    """
+
+    def __init__(self):
+        self.tmpdir = tempfile.TemporaryDirectory()
+        self.cluster_name = None
+        self.config_bundles = []
+        self.extra_config = {}
```

Comment:
I believe these three are unnecessary because they just get reassigned in `start`?

Reply:
They are referenced here: https://github.com/didithilmy/sparkconnect/blob/main/spark_connect_labextension/handlers/cluster/get_cluster_status.py

If we don't set it, `/status` will fail if the cluster is not started yet.